### PR TITLE
Fix service catalog acceptance test

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Added retry pooling loop for the broker's service classes.

**Related issue(s)**

### CODE REVIEW

File: tests/acceptance/servicecatalog/helpers_test.go [hunk 1]
```diff
@@ -62,6 +62,24 @@ func awaitCatalogContainsServiceClasses(t *testing.T, timeout time.Duration, ser
 	}
 }
 
+// awaitBrokerServiceClass awaits for service broker's not empty catalog response.
```
Comment (aszecowka) [inline tests/acceptance/servicecatalog/helpers_test.go]: can you try to use wait.AtMost function to be more consistent? Please remember that AtMost has a small bug, so ask @mszostok to copy fix from him. Additionally, maybe renaming will be required: wait.ForFuncAtMost?

Comment (polskikiel) [reply tests/acceptance/servicecatalog/helpers_test.go]: Ok, I will sync with @mszostok and align this solution to use shared waiter.

File: tests/acceptance/servicecatalog/servicecatalog_test.go [hunk 2]
```diff
@@ -56,11 +63,16 @@ func TestServiceCatalogContainsRemoteEnvironmentBrokerServiceClass(t *testing.T)
 
 		deleteClusterServiceClassesForRemoteEnvironment(t, re)
 	}()
-	brokerServices = getServiceBrokerServices(t, os.Getenv(remoteEnvironmentBrokerURLEnvName))
 
 	// then
-	assert.NotEmpty(t, brokerServices)
-	// timeout must be greater than the broker relistDuration
+	wait.ForFuncAtMost(t, func() error {
```
Comment (aszecowka) [inline tests/acceptance/servicecatalog/servicecatalog_test.go]: this test is a little bit different than for helm broker, here we create new RE and we should not only check length but if the list contains newly created RE.
Additionally, I would recommend renaming method: getServiceBrokerServices to getCatalogForBroker

File: tests/acceptance/servicecatalog/wait/wait.go [hunk 3]
```diff
@@ -5,7 +5,7 @@ import (
 	"time"
 )
 
-func AtMost(t *testing.T, fn func() error, timeout time.Duration) {
+func ForFuncAtMost(t *testing.T, fn func() error, timeout time.Duration) {
```
Comment (aszecowka) [inline tests/acceptance/servicecatalog/wait/wait.go]: this function has a flaw, I asked you to fixed it.

**Pull request merged**